File: ProtoSAS/src/main/java/dd/protosas/computation/LevelHolder.java
package dd.protosas.computation;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by Sergey on 20.01.2016.
 */
public class LevelHolder {

    private List<LightweightLevel> levels = new ArrayList<>();

    public void process(){
        levels.forEach(LightweightLevel::process);
    }

    public void addLevel(LightweightLevel level){
        levels.add(level);
    }

    public LightweightLevel getLevel(int levelNum){
        return levels.get(levelNum);
    }
}

File: ProtoSASOnScala/src/main/scala/dd/sas/pipeline/calculation/structures/LevelHolder.scala
package dd.sas.pipeline.calculation.structures

import scala.collection.mutable.ListBuffer


/**
  * Created by Sergey on 11.10.2016.
  */
class LevelHolder {
  private var levels: ListBuffer[Level] = ListBuffer[Level]()

  def process(): Unit = {
    println("levels = " + levels.size)
    levels foreach (_.process)
  }

  def addUpdatableLevels(level: Level): Unit = levels += level
}
